### PR TITLE
fix(Scripts/Spells): Implement damage scaling based off DoT stack count for Seal of Vengeance and Corruption.

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).
-  [X] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/6566

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [X] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue

## Known Issues and TODO List:

- [ ] Behavior with Seals of the Pure should be double checked.
- [ ] DoT damage scaling might also want to be investigated.
- [ ] This script applies to two spells with rather different names, no idea if the approach used is correct.
Neither of these issues will be tackled by this PR. I'm sick of spell script already.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -7833,19 +7833,17 @@ bool Unit::HandleDummyAuraProc(Unit* victim, uint32 damage, AuraEffect* triggere
...
                                 {
                                     if (stacker)
                                         aur->RefreshDuration();
-
-                                    CastSpell(victim, 42463, true, castItem, triggeredByAura);
-                                    return true;
                                 }
                             }
 
+                            CastSpell(victim, 42463, true, castItem, triggeredByAura);
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:7845]: you move it you fix it. spell id

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -7866,19 +7864,17 @@ bool Unit::HandleDummyAuraProc(Unit* victim, uint32 damage, AuraEffect* triggere
...
                                 {
                                     if (stacker)
                                         aur->RefreshDuration();
-
-                                    CastSpell(victim, 53739, true, castItem, triggeredByAura);
-                                    return true;
                                 }
                             }
 
+                            CastSpell(victim, 53739, true, castItem, triggeredByAura);
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:7876]: same

File: src/server/scripts/Spells/spell_paladin.cpp [hunk 3]
```diff
@@ -1105,6 +1109,40 @@ class spell_pal_seal_of_righteousness : public AuraScript
...
+{
+    PrepareSpellScript(spell_pal_seal_of_vengeance);
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Unit* target = GetExplTargetUnit();
+        uint32 spellId = GetSpell()->GetSpellInfo()->Id;
+        uint32 auraId = (spellId == SPELL_PALADIN_SEAL_OF_VENGEANCE_EFFECT)
+            ? SPELL_PALADIN_HOLY_VENGEANCE
+            : SPELL_PALADIN_BLOOD_CORRUPTION;
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_paladin.cpp:1124]: Validate the spells

Comment (heyitsbench) [reply src/server/scripts/Spells/spell_paladin.cpp:1130]: Unfamiliar with spell scripts, not sure if I did it right.

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (heyitsbench) [reply src/server/game/Entities/Unit/Unit.cpp]: Ain't no way you wanted me to set up enums in Unit.cpp for specific spells, so I added comments for the spell names. I hope that's what you wanted.

Comment (Kitzunu) [reply src/server/game/Entities/Unit/Unit.cpp]: Yes

File: src/server/scripts/Spells/spell_paladin.cpp (hunk 3 above)
Comment (Kitzunu) [reply src/server/scripts/Spells/spell_paladin.cpp:1130]: Yup. Rule of thumb is that if a spell is used within a script it has to be validated

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**